### PR TITLE
Add client protocol for member demotion [MC-2855]

### PR DESCRIPTION
**Pull request by emrberk**

### CODE REVIEW

**Review by promanenko — APPROVED** [review]

**Review by userMuratt — APPROVED** [review]

File: protocol-definitions/MC.yaml [hunk 1]
```diff
@@ -955,3 +955,12 @@ methods:
           doc: |
             true if the session is found & closed,
             false otherwise.
+  - id: 37
+    name: demoteDataMember
+    since: 2.8
+    doc: |
+      Demotes the data member it's called on to a lite member.
+    request:
+      retryable: false
```
Comment (ihsandemir) [inline protocol-definitions/MC.yaml:964]: why not retryable? is it not idempotent?

File: protocol-definitions/MC.yaml [hunk 2]
```diff
@@ -955,3 +955,12 @@ methods:
           doc: |
             true if the session is found & closed,
             false otherwise.
+  - id: 37
+    name: demoteDataMember
```
Comment (ihsandemir) [inline protocol-definitions/MC.yaml:959]: can we have a more descriptive name if it is demoting to a lite member, e.g.  `changeToLiteMember` or `demoteToLiteMember` ?

File: protocol-definitions/MC.yaml [hunk 3]
```diff
@@ -955,3 +955,12 @@ methods:
...
+    since: 2.8
+    doc: |
+      Demotes the data member it's called on to a lite member.
+    request:
+      retryable: false
+      partitionIdentifier: -1
+    response: {}
```
Comment (ihsandemir) [inline protocol-definitions/MC.yaml:966]: how do we communicate success and failure? If communicate failure with exceptions, can you document what are possible exceptions as well?

File: protocol-definitions/MC.yaml (hunk 1 above)
Comment (emrberk) [reply protocol-definitions/MC.yaml:964]: Should we retry demoting the member after exception is thrown? I thought it won't add up any value.

File: protocol-definitions/MC.yaml (hunk 2 above)
Comment (promanenko) [reply protocol-definitions/MC.yaml:959]: it's consistent with the existing `promoteLiteMember`

File: protocol-definitions/MC.yaml (hunk 1 above)
Comment (promanenko) [reply protocol-definitions/MC.yaml:964]: we should not retry it

File: protocol-definitions/MC.yaml (hunk 3 above)
Comment (emrberk) [reply protocol-definitions/MC.yaml:967]: Exception is thrown in case of a failure. Added to the doc part.

**Review by zhmaeff — APPROVED** [review]

**Pull request merged**